core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -282,6 +282,9 @@ private[spark] class SecurityManager(sparkConf: SparkConf)
       }: TrustManager
     })
 
+    require(fileServerSSLOptions.protocol.isDefined,
+      "spark.ssl.protocol is required when enabling SSL connections.")
+
     val sslContext = SSLContext.getInstance(fileServerSSLOptions.protocol.getOrElse("Default"))
```

Comment:
OK, though "getOrElse" is redundant now. "Default" is a meaningless value and I thought that was what the JIRA wanted to also remove.